src/roles/common/tasks/check-predeploy-prereq.yml

```diff
@@ -71,6 +71,24 @@
 
   when: not vsd_inplace_upgrade | default(false)
 
+- name: Check for portal files missing
+  set_fact:
+    is_portal_path_missing: "{{ portal_path | default('MISSING') == 'MISSING' }}"
+    is_portal_file_missing: "{{ portal_file_name | default('MISSING') == 'MISSING' }}"
+
+- name: Delete the md5 files
+  file:
+    path: "{{ inventory_dir }}/deployment.md5.yml"
+    state: absent
+  when: is_portal_file_missing or is_portal_path_missing
+  delegate_to: localhost
+
+- name: Check the image path for portal container
+  assert:
+    that: not is_portal_file_missing or not is_portal_path_missing
```

Comment:
logic needs to be:

not is_portal_file_missing and not is_portal_path_missing

Reply:
We want to fail if any one file is missing correct

Reply:
Yes, but correct logic is:
not is_portal_file_missing and not is_portal_path_missing

De Morgan's law:
not (A or B) = not A and not B
not (A and B) = not A or not B

(First one applies here)

Reply:
yeah you are correct. Done